jupyter-extension/jupyterlab_pachyderm/pfs_manager.py

```diff
@@ -479,24 +506,65 @@ def __init__(self, client: Client, **kwargs):
                 f"Could not create mount dir {PFS_MOUNT_DIR}. Try setting the PFS_MOUNT_DIR env var."
             )
             raise e
-        super().__init__(**kwargs)
 
-    # TODO: don't ignore name in the input spec
-    # right now, when a repo is mounted as part of mounting datum(s), the
-    # toplevel directory for the repo is project_repo_branch. we should
-    # update this to respect the name in the pfsinput objects being supplied
-    # to the mount_datums api. however, this is not that simple as the name
-    # can't simply be looked up in the datuminfo. we will need to have some
-    # mapping to track which files within a datum belong to which name.
+    def _populate_name_table(self, input: pps.Input):
+        """
+        Given the input spec, populates the bi-directional name to commit_uri
+        mapping of the commits mounted by the input.
+        """
+        if input.pfs:
+            repo = pfs.Repo(
+                name=input.pfs.repo,
+                project=pfs.Project(
+                    name=input.pfs.project if input.pfs.project else "default"
+                ),
+            )
+            branch = pfs.Branch(
+                repo=repo, name=input.pfs.branch if input.pfs.branch else "master"
+            )
+            commit_id = (
+                input.pfs.commit
+                if input.pfs.commit
+                else self._client.pfs.inspect_branch(branch=branch).head.id
+            )
+            commit = pfs.Commit(repo=repo, id=commit_id, branch=branch)
+            commit_uri = commit.as_uri()
+            if commit_uri in self._repo_names:
+                raise ValueError(
+                    "Loading multiple instances of the same commit is currently not supported in the extension"
+                )
+            name = (
+                f"{commit.repo.project.name}_{commit.repo.name}_{commit.branch.name}"
```

Comment:
sorry i'm not sure i'm following. this line is so that we have a default name to fall back on if no name alias is provided by the user. how does this relate to name collisions?

that does remind me though, i don't think this impl handles name collisions by the user. i will add in handling for that.

Reply:
I mean do we need to keep the default as `project_repo_branch`. This seems like a good opportunity to move towards how (in a pipeline) the worker binary defaults the name to just `repo`. We should be able to do that here since we can detect name collisions and alert the user?